src/components/TxLyraSign.js

```diff
@@ -19,30 +19,72 @@ const signWithLyra = async (xdr, setTxResult) => {
     console.error(e);
   }
 
-  setTxResult(res.signedTransaction);
+  if (res.error) {
+    setTxResultMessage(res.error);
+    setErrorStatus(true);
+  } else {
+    setTxResultMessage(res.signedTransaction);
+    setErrorStatus(false);
+  }
+};
+
+const ErrorResultDetails = ({ resultMessage, hasError }) => {
+  const [isClicked, setOnClick] = React.useState(false);
+
+  let arrowIconType = isClicked
+    ? "s-button__dropdown-icon--right"
+    : "s-button__dropdown-icon";
+
+  return (
+    <div>
+      <button
+        type="button"
+        className={arrowIconType}
+        onClick={() => setOnClick(!isClicked)}
+      >
+        Details
+      </button>
+      {isClicked ? (
+        <div className="TxLyraSign__result-error">
+          <code>{resultMessage}</code>
+        </div>
+      ) : null}
+    </div>
+  );
 };
 
 const TxLyraSign = ({ xdr }) => {
-  const [txResult, setTxResult] = React.useState();
+  const [txResultMessage, setTxResultMessage] = React.useState("");
+  const [hasError, setErrorStatus] = React.useState(false);
 
   return (
     <div>
-      <hr />
       <button
         className="s-button"
         type="button"
-        onClick={() => signWithLyra(xdr, setTxResult)}
+        onClick={() => signWithLyra(xdr, setTxResultMessage, setErrorStatus)}
       >
-        Sign with Lyra
+        {txResultMessage ? "Sign again" : "Sign with Lyra"}
       </button>
-      {txResult ? (
-        <div>
-          <CodeBlock
-            className="AccountCreator__spaceTop"
-            code={JSON.stringify(txResult, null, 2)}
-            language="json"
-          />
-        </div>
+      {txResultMessage ? (
+        hasError ? (
+          <div className="TxLyraSign__result">
+            There was an error signing the transaction:
+            <ErrorResultDetails
+              resultMessage={txResultMessage}
+              hasError={hasError}
+            />
+          </div>
```

Comment:
If we only ever get a single string back, not more complex details, maybe this should just display the string instead of a hider element? I was thinking the hidden bit would be nice for long, obscure information like a stack trace.

Reply:
Makes sense @vcarl

Reply:
Updated!
<img width="624" alt="updated-to-string" src="https://user-images.githubusercontent.com/3912060/93633819-914b7c80-f9bd-11ea-9442-248a68331a39.png">